### DISCUSSION TITLE
TTL adapter without DTR or RTS pins

### DISCUSSION THREAD

**Discussion by gioele-antoci** [emporia-vue-local/esphome#258] [Q&A] [answered]
>If your TTL adapter has both the DTR and RTS pins exposed, you can let it automatically reboot the board and put the chip into flash mode when necessary. IO0 connects to DTR, and EN connects to RTS. In this case, you don't need to hold anything down.

Reading the docs I see that I would need these pins. The adapter I own (pic below) doesn't expose these pins. does it mean I can't flash esphome with this adapter or do I need to hold something down? what exactly?

![613vDNZkVZL _AC_SL1200_](https://github.com/emporia-vue-local/esphome/assets/18650836/5646ae7d-c9d2-423b-a8f7-d95c43f2bc8c)

**Comment by flaviut** [comment, marked as answer]
Scroll up a bit, there's a detailed procedure for this situation :)

**Reply by gioele-antoci** [reply]
sorry :) I hadn't realized that this part
>Hold the modified end of the cable in IO0 to the metal shield on the ESP32. If you'd like, you can tape it down so that you have both hands free.
>.... 
>While holding it in place, connect 5V on your UART adapter to the VCC_5V0 pin on the board.

was as an alternative to 

>If your TTL adapter has both the DTR and RTS pins exposed, you can let it automatically reboot the board and put the chip into flash mode when necessary. IO0 connects to DTR, and EN connects to RTS. In this case, you don't need to hold anything down.
